units/host_status.go

```diff
@@ -168,6 +169,14 @@ func setCloudHostStatus(ctx context.Context, m cloud.Manager, h host.Host, hostS
 		})
 		return errors.Wrap(m.TerminateInstance(ctx, &h, evergreen.User, "cloud provider reported host failed to start"), "error terminating instance")
 	case cloud.StatusRunning:
+		if err := initialSetup(ctx, m, &h); err != nil {
+			return errors.Wrap(err, "problem doing initial setup")
+		}
+
+		// We're done provisioning userdata hosts. The userdata script will do the rest.
+		if h.Distro.BootstrapSettings.Method == distro.BootstrapMethodUserData {
+			return errors.Wrapf(h.SetProvisionedNotRunning(), "error marking host %s as provisioned", h.Id)
+		}
```

Comment:
I know it's not exactly in the ticket, but can we move the distro.BootstrapMethodNone case in the setup host job here? That way, host.create hosts could be set to running immediately after this job finishes, since it also just requires setting the DNS name.

Reply:
done